stdlib/public/Cxx/CxxDictionary.swift

```diff
@@ -51,6 +51,20 @@ public protocol CxxDictionary<Key, Value> {
 }
 
 extension CxxDictionary {
+  /// Creates a C++ map containing the elements of a Swift Dictionary.
+  ///
+  /// This initializes the map by copying every key and value of the dictionary.
+  ///
+  /// - Complexity: O(*n*), where *n* is the number of entries in the Swift
+  ///   dictionary
+  @inlinable
+  public init(_ dictionary: Dictionary<Key, Value>) where Key: Hashable {
+    self.init()
+    for (key, value) in dictionary {
+      self[key] = value
```

Comment:
As a future performance optimization, we could use `__insertUnsafe` here to skip the check for the presence of the element in `subscript`.

Reply:
Any reason not to do it in this PR? Are there any concerns about that approach?

Reply:
No strong reasons, but I wanted to add a simple benchmark first (in a different PR) to see if the unsafe APIs actually give us any speedup compared to the subscript.